cpp/src/join/mixed_join_kernels.cu

```diff
@@ -35,18 +35,19 @@ namespace detail {
 namespace cg = cooperative_groups;
 
 template <cudf::size_type block_size, bool has_nulls>
-__global__ void mixed_join(table_device_view left_table,
-                           table_device_view right_table,
-                           table_device_view probe,
-                           table_device_view build,
-                           row_equality const equality_probe,
-                           join_kind const join_type,
-                           cudf::detail::mixed_multimap_type::device_view hash_table_view,
-                           size_type* join_output_l,
-                           size_type* join_output_r,
-                           cudf::ast::detail::expression_device_view device_expression_data,
-                           cudf::size_type const* join_result_offsets,
-                           bool const swap_tables)
+__launch_bounds__(block_size) __global__
```

Comment:
So the `block_size` bound was there before but it was unused (by mistake?), right?

Reply:
No that’s just a template argument. The directive we added is a compiler hint. Did I understand your comment?

Reply:
Oh I was wrong. `block_size` is actually used below in the code, not unused.